build/lib/src/generate/expected_outputs.dart

```diff
@@ -42,59 +42,74 @@ extension on AssetId {
 }
 
 abstract class _ParsedBuildOutputs {
-  static final RegExp _captureGroup = RegExp('{{}}');
+  static final RegExp _captureGroup = RegExp('{{([\\w\\d]*)}}');
 
   _ParsedBuildOutputs();
 
   factory _ParsedBuildOutputs.parse(
       Builder builder, String input, List<String> outputs) {
-    final groups = _captureGroup.allMatches(input).iterator;
+    final groups = _captureGroup.allMatches(input);
```

Comment:
```suggestion
    final groups = _captureGroup.allMatches(input).toList();
```

Otherwise we end up actually performing the match multiple times. The work actually happens late in the [iterator](https://github.com/dart-lang/sdk/blob/master/sdk/lib/_internal/vm/lib/regexp_patch.dart#L373), so calling methods that grab a new iterator causes it to redo all the logic each time.

I confirmed this with [a benchmark](https://gist.github.com/jakemac53/e62fb721f6cf122099c0d818af2742ff) as well and creating the list is still more efficient, although there may be a way to restructure the method so that we don't have to create the list either, but probably not a big deal.

Reply:
I think we should consider renaming this variable `matches`